docs/components/page_shell.js

```diff
@@ -17,22 +37,177 @@ if (typeof window !== 'undefined' && window.initializeMapboxAnalytics) {
     });
 }
 
+const slugger = new GithubSlugger();
+
 class PageShell extends React.Component {
 
     render() {
+        const { frontMatter, location, children } = this.props;
+        let activeTab = location.pathname.split('/')[2];
+        if (activeTab === 'example') activeTab = 'examples';
+        let topNavContent = (
+            <TopNavTabs 
+                activeTab={activeTab}
+            />
+        );
+        let productName = 'Mapbox GL JS';
+
+        let sidebarContent = <div />;
+        let sidebarStackedOnNarrowScreens = false;
+        if (location.pathname.indexOf('overview') > -1) {
```

Comment:
hmm it would be nice to abstract this if/else situation away a bit. Could we save the current tab as part of state and then have a function that accepts the current tab value, and calls a function within the appropriate data file to return the sidebar content? open to other ideas, too 💭 happy to pair on this if ya want too.

Reply:
> happy to pair on this if ya want too
Yes please! I'll put some time on the cal next week. 👯